build.gradle

```diff
@@ -29,4 +29,11 @@ wrapper {
     gradleVersion = '5.4.1'
 }
 
+task writeAgentProperties(type: WriteProperties) {
```

Comment:
it can be done easier
```groovy
processResources {
    filesMatching('agent.properties') {
        expand(project.properties)
    }
```

and in properties file:
```properties
agent.name=${name}
agent.version=${version}
```

Reply:
updated